src/Featurizers/ForwardFillImputerFeaturizer.h

```diff
@@ -0,0 +1,185 @@
+        }
+    }
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \typedef       ForwardFillImputerEstimator
+///  \brief         Estimator that creates `ForwardFillImputerTransformer`.
+///
+template <typename T>
+using ForwardFillImputerEstimator           = Components::InferenceOnlyEstimatorImpl<ForwardFillImputerTransformer<T>>;
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+template <typename T>
+ForwardFillImputerTransformer<T>::ForwardFillImputerTransformer(T defaultValue) :
+    _lastValue(std::move(defaultValue)),
+    _validLastValue(true) {
+}
+
+template <typename T>
+ForwardFillImputerTransformer<T>::ForwardFillImputerTransformer(Archive &ar) :
+    ForwardFillImputerTransformer(
+        [&ar](void) {
+            // Version
+            std::uint16_t                   majorVersion(Traits<std::uint16_t>::deserialize(ar));
+            std::uint16_t                   minorVersion(Traits<std::uint16_t>::deserialize(ar));
+
+            if(majorVersion != 1 || minorVersion != 0)
+                throw std::runtime_error("Unsupported archive version");
+
+            // Data
+            bool                            validLastValue(Traits<bool>::deserialize(ar));
+
+            if(validLastValue == false)
+                return ForwardFillImputerTransformer();
+
+            T                               lastValue(Traits<T>::deserialize(ar));
+
+            return ForwardFillImputerTransformer(std::move(lastValue));
+        }()
+    ) {
+}
+
+template <typename T>
+bool ForwardFillImputerTransformer<T>::operator==(ForwardFillImputerTransformer const &other) const {
+    if(_validLastValue != other._validLastValue)
+        return false;
+
+    if(_validLastValue == false)
+        return true;
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#endif
+
+    return _lastValue == other._lastValue;
+
+#if (defined __clang__)
+#   pragma clang diagnostic pop
+#endif
+}
+
+template <typename T>
+bool ForwardFillImputerTransformer<T>::operator!=(ForwardFillImputerTransformer const &other) const {
+    return (*this == other) == false;
+}
+
+template <typename T>
+void ForwardFillImputerTransformer<T>::save(Archive &ar) const /*override*/ {
+    // Version
+    Traits<std::uint16_t>::serialize(ar, 1); // Major
+    Traits<std::uint16_t>::serialize(ar, 0); // Minor
+
+    // Data
+    Traits<bool>::serialize(ar, _validLastValue);
+
+    if(_validLastValue)
+        Traits<T>::serialize(ar, _lastValue);

```

Comment:
I think may have a potential bug here. I believe these are being serialized so that the default value will be preserved, is that correct? If so, if I run the transformer through some data first, the `_validLastValue` will be overwritten. Then, if I save the transformer after that data has been used, it will serialize a different value then what was originally provided. I think we should go with the approach that the backfill is using, with a different variable to store just the default value.

Reply:
Per our conversation, I've updated serialization and equality to only operator on initial state (and ignore working state).